### ISSUE TITLE
Review disabling tests workflow

### ISSUE THREAD

**Issue by mruberry** [pytorch/pytorch#68498]
Today in PyTorch tests can be disabled using issues, which is super quick and convenient.

However, this means that to understand which tests are run developers must check the test itself and see if there's an issue disabling it. It's also a little confusing what the development workflow is for resolving these skips (although we could probably educate developers about how to enable the test on their PRs where appropriate), and what happens when the issue goes away and developer PRs are based on builds without the fix (although GH Actions auto-rebases PRs, which minimizes the impact of this issue).

Ideally there'd be a mechanism as simple as filing an issue that would automagically create a skip next to the test. That way the code would be a readable source of truth that developers could understand and change as usual. Automatically adding these skips can sometimes be complicated, however. We could have OnCalls add the skips manually, or we could investigate a compromise solution where PRs adding test skips are autogenerated but they modify an existing list of tests to skip instead of annotating tests directly. 

I imagine different people will have different opinions about what's best (issues, a list, manually adding skips, some other thing). PersonallyI think we should just have OnCalls and/or test owners quickly land test skips or start by filing an issue disabling a test and then convert it to an actual test skip in the code for simplicity. This is a little more work on the OnCall but means we won't have to build and maintain new infra, and as flakiness goes down we should be bothered by this less and less.

**Comment by ngimel** [comment]
I agree, OnCall should be responsible for properly skipping tests. We currently have 58 open DISABLED issues, some of them dating back to beginning of 2020, so it's not a big load.
